pkg/controller/master/rancher/embedded_rancher.go

```diff
@@ -33,6 +44,48 @@ func (h *Handler) RancherSettingOnChange(key string, setting *rancherv3api.Setti
 	return nil, nil
 }
 
+func (h *Handler) initializeServerURL(setting *rancherv3api.Setting) (*rancherv3api.Setting, error) {
+	vipConfig, err := h.getVipConfig()
+	if err != nil {
+		return nil, err
+	}
+	toUpdate := setting.DeepCopy()
+	toUpdate.Value = fmt.Sprintf("https://%s", vipConfig.IP)
+	logrus.Debugf("Updating server-url setting to %s", toUpdate.Value)
+	return h.RancherSettings.Update(toUpdate)
+}
+
+func (h *Handler) initializeSystemNamespaces(setting *rancherv3api.Setting) (*rancherv3api.Setting, error) {
+	sets := labels.Set{
+		defaultAdminLabelKey: defaultAdminLabelValue,
+	}
+	users, err := h.RancherUserCache.List(sets.AsSelector())
+	if err != nil {
+		return nil, err
+	}
+	if len(users) == 0 {
+		// The default admin and its namespace is not created, enqueue and check later
+		h.RancherSettingController.EnqueueAfter(setting.Name, 30*time.Second)
+		return nil, nil
+	}
+	defaultAdminNamespace := users[0].Name
+	harvesterDefaultSystemNamespaces := strings.Join([]string{
+		ranchersettings.SystemNamespaces.Default,
+		"cattle-dashboards",
+		"cattle-fleet-clusters-system",
+		"cattle-monitoring-system",
```

Comment:
Do u think if we use `cattle-*` as a general filter?

Reply:
No. The default system namespaces list in Rancher uses full name of each namespace. And we don't want to prevent cattle lovers from using namespaces with this pattern :smiley: